frontend/src/components/BackgroundEffects/EffectOptionButtons/EffectOptionButtons.spec.tsx

```diff
@@ -0,0 +1,44 @@
+import { describe, expect, it, vi } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import EffectOptionButtons from './EffectOptionButtons';
+
+describe('EffectOptionButtons', () => {
+  it('renders all effect options', () => {
+    render(<EffectOptionButtons backgroundSelected="none" setBackgroundSelected={() => {}} />);
+    expect(screen.getByTestId('BlockIcon')).toBeInTheDocument();
+    expect(screen.getAllByTestId('BlurOnIcon')).toHaveLength(2);
+  });
+
+  it('marks the selected option as selected', () => {
+    render(<EffectOptionButtons backgroundSelected="low-blur" setBackgroundSelected={() => {}} />);
+    const selectedOption = screen.getByTestId('background-low-blur');
+    expect(selectedOption).toBeInTheDocument();
+  });
+
+  it('calls setBackgroundSelected when an option is clicked', async () => {
+    const setBackgroundSelected = vi.fn();
+    render(
+      <EffectOptionButtons
+        backgroundSelected="none"
+        setBackgroundSelected={setBackgroundSelected}
+      />
+    );
+    const lowBlur = screen.getByTestId('background-low-blur');
+    await userEvent.click(lowBlur);
+    expect(setBackgroundSelected).toHaveBeenCalledWith('low-blur');
+  });
+
+  it('calls setBackgroundSelected with "high-blur" when high blur option is clicked', async () => {
```

Comment:
nit: the name of the test is too literal. I'd change to something like `sets the selected background with high blue when foo`